suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -709,17 +701,23 @@ export const createDiscoveryThunk = createThunk(
             selectors: { selectEnabledNetworks },
         } = extra;
         const enabledNetworks = selectEnabledNetworks(getState());
-        const networksSymbols = filterUnavailableNetworksCompatible(enabledNetworks, device).map(
-            n => n.symbol,
+        const filteredNetworks = filterUnavailableNetworks(enabledNetworks, device);
+        const networksSymbols = filteredNetworks.map(n => n.symbol);
+
+        // calculate theoretical limit of accounts per enabled networks (then `calculateProgress` will gradually converge on the real number)
+        const availableConfigAccounts = filteredNetworks.reduce(
+            (count, network) => count + filterUnavailableAccountTypes(network).length,
+            0,
         );
+        const maxTotalAccounts = LIMIT * availableConfigAccounts;
 
         dispatch(
             createDiscovery({
                 deviceState,
                 authConfirm: !device.useEmptyPassphrase,
                 index: 0,
                 status: DiscoveryStatus.IDLE,
-                total: LIMIT * networksSymbols.length,
+                total: maxTotalAccounts,
```

Comment:
I suspect this is not used anywhere? But I'm reaaally not sure. I updated it anyway.